jdaviz/configs/rampviz/plugins/viewers.py

```diff
@@ -0,0 +1,140 @@
+import numpy as np
+from astropy.nddata import NDDataArray
+from glue.core import BaseData
+from glue_jupyter.bqplot.image import BqplotImageView
+
+from jdaviz.configs.default.plugins.viewers import JdavizViewerMixin, JdavizProfileView
+from jdaviz.configs.cubeviz.plugins.mixins import WithSliceSelection, WithSliceIndicator
+from jdaviz.core.registries import viewer_registry
+from jdaviz.core.freezable_state import FreezableBqplotImageViewerState
+
+__all__ = ['RampvizProfileView', 'RampvizImageView']
+
+
+@viewer_registry("rampviz-profile-viewer", label="Profile 1D (Rampviz)")
+class RampvizProfileView(JdavizProfileView, WithSliceIndicator):
+    # categories: zoom resets, zoom, pan, subset, select tools, shortcuts
+    tools_nested = [
+                    ['jdaviz:homezoom', 'jdaviz:prevzoom'],
+                    ['jdaviz:boxzoom', 'jdaviz:xrangezoom', 'jdaviz:yrangezoom'],
+                    ['jdaviz:panzoom', 'jdaviz:panzoom_x', 'jdaviz:panzoom_y'],
+                    ['jdaviz:selectslice'],
+                    ['jdaviz:sidebar_plot', 'jdaviz:sidebar_export']
+                ]
+
+    default_class = NDDataArray
+    _default_profile_subset_type = 'temporal'
+
+    def __init__(self, *args, **kwargs):
+        kwargs.setdefault('default_tool_priority', ['jdaviz:selectslice'])
+        super().__init__(*args, **kwargs)
+
+    def _initialize_x_axis(self):
+        if len(self.state.x_att_helper.choices):
+            self.state.x_att = self.state.x_att_helper.choices[-1]
+            self.set_plot_axes()
+            self.reset_limits()
+
+    def reset_limits(self):
+        super().reset_limits()
+
+        # override to reset to the global y limits including marks:
```

Comment:
this is inconsistent with the rest of jdaviz - is this something we should consider moving up higher to be shared everywhere?

Reply:
I agree, it's intentionally inconsistent. The assumption elsewhere is that that loaded `Data` entries set the axis limits. The ramp viewer is most useful when you're creating and editing subsets to inspect single-pixel ramps, which aren't loaded Data entries until you extract them. 

Before I made this override, it was very difficult to see the relevant dynamic range of the single-pixel ramps as you moved onto and off of a source, since the y-axis values can range from 10^2 to 10^4. Zooming in and out using the y-zoom requires a few clicks and a lot of experimentation to get right. This was change is an improvement in this viewer.

We could use a similar change in Cubeviz if others agreed that the spectrum-viewer's axis limits should update when adding a subset. I'm in favor!

Reply:
ok, the override here is ok for now then and let's just remember this if there ever is a request for anything similar in cubeviz to then generalize instead of creating a separate behavior?